.github/workflows/CI.yml

```diff
@@ -0,0 +1,103 @@
+name: CI
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  # pre-commit:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v4
+  #     - uses: actions/setup-python@v5
+  #     - uses: pre-commit/action@2c7b3805fd2a0fd8c1884dcaebf91fc102a13ecd  # v3.0.1
+
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      tracked-files: ${{ steps.filter.outputs.tracked-files }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Filter files
+        uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 #v3.0.2
+        id: filter
+        with:
+          filters: |
+            tracked-files:
+              - 'setup.py'
+              - 'pyproject.toml'
+              - '.conda/**'
+              - 'src/**'
+
+  verify-conda-build:
+    name: Verify Conda Build
+    runs-on: ubuntu-latest
+    needs: changes
+    # Only run if there are changes in the tracked files
+    if: ${{ needs.changes.outputs.tracked-files == 'true' }}
```

Comment:
Just a styling thing, but you don't need the `${{ }}` bit in an `if` section, strangely enough.

Reply:
I know, but I like to keep them for consistency with other cases where you need them (for example when you have multiple conditions with AND, OR logical connectors).